AI #2334: Align remaining normative requirement inconsistencies with updated guidelines (#2344)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/invoice_detail/columns/invoiceissuestatus.md

* InvoiceIssueStatus MUST NOT be null.
* InvoiceIssueStatus MUST be one of the allowed values.
* InvoiceIssueStatus MUST represent the current publication state of the invoice.

> * InvoiceIssueStatus MUST NOT be "Open" when InvoiceIssueStatus was previously "Issued", unless explicitly requested or approved by the customer.

Author: @shawnalpay, following the pattern you suggested for `BillingPeriodStatus`, I’d suggest aligning this requirement as well.

> * BillingPeriodStatus MUST NOT be "Open" following a "Closed" state, except when explicitly requested or approved by the customer.

Additionally, I suggest we:
- use `status` instead of `state` (to align with the column name)
- remove the article (`a`/`an`), since this refers to a specific enum value rather than a generic state

Thoughts?

```suggestion
* InvoiceIssueStatus MUST NOT be "Open" following "Issued" status, except when explicitly requested or approved by the customer.
```

Reviewer: I don't prefer cutting the article, as it makes it less English readable.  Changing to "status" is okay with me.

I feel like we need to add "previous" here as well.  Want to keep it very clear that "Issued" was the previous state.

```suggestion
* InvoiceIssueStatus MUST NOT be "Open" following a previous status of "Issued", except when explicitly requested or approved by the customer.
```
